catfs/fs.go

```diff
@@ -1021,6 +985,16 @@ func (fs *FS) Stage(path string, r io.ReadSeeker) error {
 		return err
 	}
 
+	// The stream was consumed, we now know those attrs:
+	size := sizeAcc.Size()
+	contentHash := hashWriter.Finalize()
+	compressAlgo, err = compress.GuessAlgorithm(path, headerReader.Header())
```

Comment:
We should run guessing before fs.bk.Add, here it useless and potentially incorrect.
By this time we already assigned default `comressAlgo` and used it with `mio` in `fs.bk.Add` couple lines above.

Reply:
Right, I had a logic error here. Now I remember why ``PeekHeader()`` actually did the reading before the add. :joy:
Seems I need to use the existing ``prefixReader`` but patch the part I read to the stream. Luckily Go makes that easy, that would be quite a horror in C/C++. Too bad I wrote some nice tests for ``HeaderReader``.